### PR TITLE
fix: improve video scaling

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: app/script/components/groupVideoGrid.js [hunk 1]
```diff
@@ -58,31 +58,59 @@ z.components.GroupVideoGrid = (() => {
...
+
       this.minimized = params.minimized;
     }
 
-    scaleVideos(elements) {
+    scaleVideos(rootElement) {
+      const elements = Array.from(rootElement.querySelectorAll('.group-video-grid__element'));
       elements
         .filter(element => !!element.classList)
         .filter(element => element.classList.contains('group-video-grid__element'))
```
Comment (AndyLnd) [inline app/script/components/groupVideoGrid.js]: No need to filter them by there class name as you already selected only those with that class.

File: app/script/components/groupVideoGrid.js [hunk 2]
```diff
@@ -58,31 +58,59 @@ z.components.GroupVideoGrid = (() => {
...
+      this.scaleVideos = this.scaleVideos.bind(this, rootElement);
+      // scale videos when the grid is updated (on the next rendering cycle)
+      this.grid.subscribe(() => {
+        window.requestAnimationFrame(this.scaleVideos);
+      });
+
+      // scale the videos when the window is resized
+      window.addEventListener('resize', this.scaleVideos);
+
+      this.rootElement = rootElement;
```
Comment (AndyLnd) [inline app/script/components/groupVideoGrid.js]: you already bound it to the function, do you still need it somewhere else?

Comment (atomrc) [reply app/script/components/groupVideoGrid.js]: very good point. It was needed for the previous iteration and I forgot to remove it :D 
:+1:

File: app/script/components/groupVideoGrid.js (hunk 1 above)
Comment (atomrc) [reply app/script/components/groupVideoGrid.js]: Very true

File: app/script/components/groupVideoGrid.js [hunk 3]
```diff
@@ -197,8 +220,8 @@ z.components.GroupVideoGrid = (() => {
 
   ko.components.register('group-video-grid', {
     template: `
-      <div class="group-video">
-        <div class="group-video-grid" data-bind="foreach: { data: grid, as: 'streamId', afterRender: scaleVideos}">
+      <div class="group-video" data-bind="template: { afterRender: scaleVideos }">
+        <div class="group-video-grid" data-bind="foreach: { data: grid, as: 'streamId' }">
```
Comment (gregor) [inline app/script/components/groupVideoGrid.js:224]: Too much whitespace:

````
<div class="group-video" data-bind="template: {afterRender: scaleVideos}">
  <div class="group-video-grid" data-bind="foreach: {data: grid, as: 'streamId'}">
````

**Pull request merged**